coap-gateway/service/client.go

```diff
@@ -251,77 +281,95 @@ func (client *Client) cleanObservedResources() {
 	}
 }
 
+func (client *Client) cancelResourceSubscriptions(wantWait bool) {
+	resourceSubscriptions := client.resourceSubscriptions.PullOutAll()
+	for _, v := range resourceSubscriptions {
+		o := grpcClient.ToResourceSubscription(v, true)
+		wait, err := o.Cancel()
```

Comment:
`o` might be `nil`? Good practice is to use an additional `ok` return value instead of just returning `nil`.